src/tags/core.js

```diff
@@ -23,10 +23,11 @@ export const nullObj = {
     ctx.wrapScalars ? new Scalar(null) : null,
   default: true,
   tag: 'tag:yaml.org,2002:null',
+  sourceStr: null,
   test: /^(?:~|[Nn]ull|NULL)?$/,
   resolve: () => null,
   options: nullOptions,
-  stringify: () => nullOptions.nullStr
+  stringify: ({ sourceStr }) => sourceStr ?? nullOptions.nullStr
```

Comment:
+ You mean "sourceStr" is not quite suitable? How about "originalStr".
+ And, I have add the codes and testcase to reparse the string which matches the defined custom null string, 
+ But it can not handle the `"!!null str"` or`"a: \"None\""` when
> YAML.scalarOptions.null.nullStr = 'None'

I think it is normal.

Reply:
Let's go with `sourceStr` for now.

There's no need to easily support a non-YAML representation of `null` like `'None'`.

Reply:
> it should be tested with the tag's regexp to make sure that it'll re-parse as null

oops, I misunderstood it.